### ISSUE TITLE
When the user uploads a file, NVDA announces the top of the page (“banner landmark link Skip to Main Content”), but focus is actually still on the Upload Files button. Only the focused item should be announced.

### ISSUE THREAD

**Issue by Phatal900** [nvaccess/nvda#19607]
### Brief summary

Hello we are seeing an issue where when the user uploads a file, NVDA announces the top of the page (“banner landmark link Skip to Main Content”), but focus is actually still on the Upload Files button.  Only the focused item should be announced.

CCSCM-34621

### Steps to reproduce

1. Tab to the accordion section
2. Tab to the upload files button and upload a sample file. 

<img width="1636" height="575" alt="Image" src="https://github.com/user-attachments/assets/52a3b801-ad44-4a50-a431-36f942d6743f" />

### Actual behavior

The issue is that NVDA announces the top of the page (“banner landmark link Skip to Main Content”), but focus is actually still on the Upload Files button.  Only the focused item should be announced.

### Expected behavior

Only the focused item should be announced when uploading a file. 

### NVDA type

installed copy

### NVDA version

NVDA Version 2025.3.2 (2025.3.2.53588)

### Windows version

Windows 11 Enterprise23H2 Build 22631.6491

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have run the COM Registration Fixing Tool and the issue still occurs

**Comment by seanbudd** [comment]
Welcome @Phatal900,

We are unable to process this issue as it stands.

To be able to reproduce this issue, we need a code sample. Provide a minimal HTML sample to reproduce this using [codepen](https://codepen.io/).

Kind Regards,
NV Access Software Developers

**Comment by EllyLoel** [comment]
i believe this is an issue with the `<dialog>` element as it's not an issue for aria dialogs managing focus with javascript like for example [the apg one](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/dialog/). upon closing the dialog the first item in the banner landmark is announced before the item where focus is placed is announced.

i've create a [minimal example where the issue is reproducable](https://cdpn.io/pen/debug/zxKOzKQ?authentication_hash=XBAGRmNRJLyr) with NVDA 2025.2.0.37684 and any of these browsers: Firefox 147.0.3, Edge 145.0.3800.70 and Chrome 145.0.7632.77.

<img width="960" height="565" alt="A screenshot of the codepen example in Firefox with NVDA open and the speach viewer showing that the first item in the banner landmark is announced before announcing the button where focus is moved after closing the modal dialog." src="https://github.com/user-attachments/assets/af25913c-052c-415e-9628-f630943a8ce6" />

**Comment by seanbudd** [comment]
@EllyLoel - the link you sent has expired. Please use a permalink to the codepen

**Comment by EllyLoel** [comment]
ah! so sorry, here’s the codepen link: https://codepen.io/ellyloel/pen/EagYXgB
